pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -96,11 +99,43 @@ func initContainerName(lang language) string {
 	return fmt.Sprintf("datadog-lib-%s-init", lang)
 }
 
+// getDeploymentReference returns the OwnerReference if it is of Kind Deployment
+func getDeploymentReference(pod *corev1.Pod) *v1.OwnerReference {
+	ownerReferences := pod.ObjectMeta.OwnerReferences
+
+	for _, ownerRef := range ownerReferences {
+		if ownerRef.Kind == "Deployment" {
+			return &ownerRef
+		}
+	}
+	return nil
+}
+
 func injectAutoInstrumentation(pod *corev1.Pod, _ string, _ dynamic.Interface) error {
 	if pod == nil {
 		return errors.New("cannot inject lib into nil pod")
 	}
 
+	if config.Datadog.GetBool("admission_controller.auto_instrumentation.apm_instrumentation_enabled") {
```

Comment:
What would you think of enabling this feature only if `admission_controller.mutate_unlabelled ` is also `true` ?

Reply:
`mutate_unlabelled` is by default false which means that the user needs to set it to true in order to use this feature. We want as few settings possible to enable this feature to minimize onboarding/setup overhead which is why we chose to override/ignore `mutate_unlabelled` with this new setting